LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -1047,6 +1035,918 @@ public static void CopyMatToBMOorMIC(PackageEditorWindow pew)
+                {
+                    ShowError("Selected export is not a WwiseBank");
+                    return;
+                }
+
+                wwiseBankEntry = (ExportEntry)pew.SelectedItem.Entry;
+
+                newWwiseBankName = PromptDialog.Prompt(null, "New WwiseBanke name:", "New name");
+                // Check that the new name is not empty, no longe than 255, and doesn't contain white-spaces or symbols aside from _ or -
+                if (string.IsNullOrEmpty(newWwiseBankName) || newWwiseBankName.Length > 240 || newWwiseBankName.Any(c => char.IsWhiteSpace(c) || (!(c is '_' or '-') && !char.IsLetterOrDigit(c))))
+                {
+                    ShowError("Invalid name. It must not be empty, be longer than 240 characters, or contain whitespaces or symbols aside from '-' and '_'");
+                    return;
+                }
+            }
+
+            string oldBankName = wwiseBankEntry.ObjectName;
+            IntProperty bankIDProp = wwiseBankEntry.GetProperty<IntProperty>("Id");
+
+            // Get/Generate IDs and little endian hashes
+            uint oldBankID = unchecked((uint)bankIDProp.Value);
+            string oldBankHash = BigToLittleEndian(string.Format("{0:X2}", oldBankID).PadLeft(8, '0'));
+
+            uint newBankID = GetBankId(newWwiseBankName);
+            string newBankHash = BigToLittleEndian(string.Format("{0:X2}", newBankID).PadLeft(8, '0'));
+
+            // Write the replaced ID property
+            bankIDProp.Value = unchecked((int)newBankID);
+            wwiseBankEntry.WriteProperty(bankIDProp);
+
+            WwiseBank wwiseBank = wwiseBankEntry.GetBinaryData<WwiseBank>();
+            // Update the bank id
+            wwiseBank.ID = newBankID;
+
+            // Update referenced banks kvp that reference the old name
+            IEnumerable<KeyValuePair<uint, string>> updatedBanks = wwiseBank.ReferencedBanks
+                .Select(referencedBank =>
+                {
+                    if (referencedBank.Value.Equals(oldBankName, StringComparison.OrdinalIgnoreCase)) { return new(newBankID, newWwiseBankName); }
+                    else { return referencedBank; }
+                });
+            wwiseBank.ReferencedBanks = new OrderedMultiValueDictionary<uint, string>(updatedBanks);
+
+            // Update references to the old hash at the end of HIRC objects when they are found.
+            // This is mostly safe, since we know the reference appears at the end of the unparsed data,
+            // and we only replace it there.
+            foreach (WwiseBank.HIRCObject hirc in wwiseBank.HIRCObjects.Values())
+            {
+                byte[] unparsed = hirc.unparsed;
+                if (unparsed != null && unparsed.Length >= 4) // Only replace if not null and at least width of hash
+                {
+                    byte[] oldArr = Convert.FromHexString(oldBankHash);
+                    byte[] newArr = Convert.FromHexString(newBankHash);
```

Comment:
These should be hoisted out of the loop. No need to recalculate every time. 

Also, it looks like this is the only time you use these strings? If so, there's no need to have the strings at all, or to do any endianness reversing. Just use `BinaryConverter.GetBytes`.

Reply:
Yeah, having it in the loop is pretty dumb on my part, hahaha. And yeah, that's the only place. I believe I had the strings because I thought I'd need it more, or did use it more in a previous iteration, but after some tweaking this became the only usage of it